src/devices/Ili934x/Chsc6440.cs

```diff
@@ -0,0 +1,410 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Device.I2c;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Iot.Device.Ili934x
+{
+    /// <summary>
+    /// Binding for Chipsemi CHSC6540 capacitive touch screen controller
+    /// Used for instance on the M5Tough in conjunction with an ILI9342 display controller.
+    /// Note: The M5Core2, while being very similar to the M5Tough otherwise, has a FT6336U instead.
+    /// The two chips appear to be similar, but the documentation is poor.
+    /// </summary>
+    public class Chsc6440 : IDisposable
+    {
+        /// <summary>
+        /// The default I2C address of this chip
+        /// </summary>
+        public const int DefaultI2cAddress = 0x2E;
+
+        private readonly int _interruptPin;
+        private readonly bool _shouldDispose;
+        private GpioController? _gpioController;
+        private I2cDevice _i2c;
+
+        private bool _wasRead;
+
+        private TimeSpan _interval;
+
+        private Point[] _lastPoints;
+        private Point _initialTouchPoint;
+        private Point[] _points;
+        private int _point0finger;
+
+        private int _activeTouches;
+        private int _lastActiveTouches;
+        private bool _dragging;
+
+        private bool _isPressed;
+
+        private Thread? _updateThread;
+        private bool _updateThreadActive;
+        private object _lock;
+        private AutoResetEvent _updateEvent;
+
+        /// <summary>
+        /// This event is fired when the user "clicks" a position
+        /// Call <see cref="EnableEvents"/> to use event handling
+        /// </summary>
+        public event Action<object, Point>? Touched;
+
+        /// <summary>
+        /// This event is fired repeatedly when the user drags over the screen
+        /// Call <see cref="EnableEvents"/> to use event handling.
+        /// </summary>
+        public event Action<object, DragEventArgs>? Dragging;
```

Comment:
Add some prefix like "On" suggesting this is an event. Dragging suggest the type is `bool`

Reply:
(same for other events here)

Reply:
No, see e.g. https://learn.microsoft.com/en-us/dotnet/api/system.windows.window?view=windowsdesktop-7.0 It's a verb in past tense (`Touched`, `Clicked`, etc. ) for the events, and a method such as `OnTouched`, `OnClicked` for the event _handlers_.